src/tests/roc_sndio/target_sox/test_sox_source.cpp

```diff
@@ -86,12 +92,13 @@ TEST(sox_source, sample_rate_auto) {
         SoxSink sox_sink(allocator, sink_config);
         CHECK(sox_sink.open(NULL, file.path()));
 
-        Pump pump(buffer_pool, mock_source, sox_sink, FrameSize, Pump::ModeOneshot);
+        Pump pump(buffer_pool, mock_source, sox_sink, FrameLength, SampleRate, ChMask,
+                  Pump::ModeOneshot);
         CHECK(pump.valid());
         CHECK(pump.run());
     }
 
-    source_config.sample_rate = 0;
+    source_config.sample_rate = SampleRate;
```

Comment:
Unintentional change?

Reply:
I changed this because was checking for build failures in other places and this one was always failing, and later forgot to change it back to 0. I think the issue is, when sample rate is 0, frame length will become infinite. So I think I should check for this case in SoxSink and SoxSource constructors.

Reply:
```
src/modules/roc_sndio/target_sox/roc_sndio/sox_sink.cpp:63: error: roc_
panic()

ERROR: roc_sndio: !valid_
```
This is the error I get.

Reply:
If sample rate is zero, the preferred device rate is gathered from sox or pulseaudio. This is a special mode. We should keep it working. So I guess we should calculate frame size from frame length only after acquiring actual rate.